doc/design/embargo-mvp.md

```diff
@@ -0,0 +1,131 @@
+* `PRIVATE` means that the Dandiset is embargoed.
+  It is not searchable or viewable to non-owners.
+  It is not publishable.
+  Manifest YAML/JSON files will be written to the embargo bucket rather than the public bucket.
+* `RELEASING` means that the Dandiset is currently transitioning from embargoed to public.
+  All modification operations will return a 400 error while the release is in progress.
+  This includes metadata changes and uploads.
+
+A new `EmbargoedAssetBlob` model will be added.
+This behaves the same as a normal `AssetBlob`, but points to the embargo bucket rather than the public bucket.
+It also contains a reference to the `Dandiset` it belongs to.
+
+`Asset`s will point to exactly one `AssetBlob` or `EmbargoedAssetBlob`.
+An `Asset` will be considered "embargoed" if it has an `EmbargoedAssetBlob`.
+An embargoed `Asset` can only belong to the dandiset named in the `EmbargoedAssetBlob`.
+The automatically populated `Asset` metadata will also set the `access` field to the appropriate value depending on whether or not the asset is embargoed.
+
+The `Upload` model will have a new optional field `embargoed_dandiset` that points to an embargoed dandiset.
+If specified, the uploaded data will be sent to the embargoed bucket instead of the public bucket, and the `Upload` will create an `EmbargoedAssetBlob` when finalized.
+
+## API
+* Create dandiset endpoint
+
+  Add a new optional `award_number` field.
+  If set, the `access` metadata field will be set to `EmbargoedAccess` and the award number will be recorded as a `Funder` on a new Contributor Organization.
+  The `embargo_status` of the dandiset will also be set to `PRIVATE`.
+
+* Get/List dandiset endpoint
+
+  The DandisetViewSet queryset will filter out dandisets with `embargo_status == PRIVATE` that are also not owned by the current user.
+  This will prevent them from showing up in the listing or fetching endpoints.
+
+  The `DandisetSerializer` will also be updated to include the `embargo_status` field so that the web client can render the embargoed dandiset appropriately.
+
+* publish version endpoint
+
+  Return error 400 if `dandiset.embargo_status != PUBLIC`.
+
+* create asset, update metadata, and any other dandiset/version modification endpoints:
+
+  Return error 400 if `dandiset.embargo_status == RELEASING`.
+
+* New endpoint: `POST /api/dandisets/{dandiset_id}/release`
+
+  Release an embargoed dandiset.
+  
+  Only permitted for owners and admins. If the `embargo_status` is `PUBLIC` or `RELEASING`, return 400.
+
+  Set the `embargo_status` to `RELEASING`, then dispatch the release task.
+
+* Release task
+
+  For every `Asset` with an `EmbargoedAssetBlob` in the dandiset, convert the `EmbargoedAssetBlob` into an `AssetBlob` by moving the data from the embargo bucket to the public bucket.
+  These could be >5GB, so the [multipart copy API](https://docs.aws.amazon.com/AmazonS3/latest/userguide/CopyingObjectsMPUapi.html) must be used.
```

Comment:
https://aws.amazon.com/premiumsupport/knowledge-center/s3-large-transfer-between-buckets/

the main thing is to ensure we use something that doesn't change the etag.